src/pool/srv_cli.c

```diff
@@ -367,46 +368,67 @@ pool_query_init(uuid_t pool_uuid, crt_rpc_t *rpc, void *varg)
 }
 
 static int
-process_query_result(d_rank_list_t **ranks, daos_pool_info_t *info, uuid_t pool_uuid,
-		     uint32_t map_version, uint32_t leader_rank, struct daos_pool_space *ps,
+process_query_result(d_rank_list_t **enabled_ranks, d_rank_list_t **disabled_ranks,
+		     daos_pool_info_t *info, uuid_t pool_uuid, uint32_t map_version,
+		     uint32_t leader_rank, struct daos_pool_space *ps,
 		     struct daos_rebuild_status *rs, struct pool_buf *map_buf)
 {
-	struct pool_map	       *map;
-	int			rc;
-	unsigned int		num_disabled = 0;
+	struct pool_map *map                = NULL;
+	unsigned int     num_disabled       = 0;
+	d_rank_list_t   *enabled_rank_list  = NULL;
+	d_rank_list_t   *disabled_rank_list = NULL;
+	int              rc;
 
 	rc = pool_map_create(map_buf, map_version, &map);
 	if (rc != 0) {
-		D_ERROR(DF_UUID": failed to create local pool map, "DF_RC"\n",
-			DP_UUID(pool_uuid), DP_RC(rc));
-		return rc;
+		DL_ERROR(rc, DF_UUID ": failed to create local pool map", DP_UUID(pool_uuid));
+		D_GOTO(error, rc = rc);
 	}
 
 	rc = pool_map_find_failed_tgts(map, NULL, &num_disabled);
 	if (rc != 0) {
-		D_ERROR(DF_UUID": failed to get num disabled tgts, "DF_RC"\n",
-			DP_UUID(pool_uuid), DP_RC(rc));
-		goto out;
+		DL_ERROR(rc, DF_UUID ": failed to get num disabled tgts", DP_UUID(pool_uuid));
+		D_GOTO(error, rc = rc);
 	}
-	info->pi_ndisabled = num_disabled;
 
-	if (ranks != NULL) {
-		bool	get_enabled = (info ? ((info->pi_bits & DPI_ENGINES_ENABLED) != 0) : false);
+	if (info != NULL && (info->pi_bits & DPI_ENGINES_ENABLED) != 0) {
```

Comment:
Is a NULL daos_pool_info_t valid here? if so then maybe GOTO label above pool_query_reply_to_info might be useful to skip to in the case of NULL. It looks like we are using the info reference as both input and an output, if so then should we update the documentation for dsc_pool_svc_query to mark the field as in/out? It could be made more clear by reading the input bit-fields from info before explicitly specifying output values in a separate info_out. This suggestion could be considered as overkill but I've occasionally found mixing input output to be problematic.

Reply:
Yep, I agree that the usage of the info var is confusing.
Thanks for your attention.

- [ ] Refactor the info var

Reply:
> Yep, I agree that the usage of the info var is confusing. Thanks for your attention.
> 
> * [x]  Fix the `dsc_pool_svc_query() `documentation
> * [x]  Refactor the info var

Fixed with commit 87ff98c41